app/celery/tasks.py

```diff
@@ -170,3 +170,51 @@ def create_pdf_for_templated_letter(self, encrypted_letter_data):
         },
         queue=QueueNames.LETTERS
     )
+
+
+@notify_celery.task(name='recreate-pdf-for-precompiled-letter')
+def recreate_pdf_for_precompiled_letter(notification_id, folder_name, filename, allow_international_letters):
+    """
+    This task takes the details of a PDF letter which we want to recreate as its arguments.
+    It gets the backup version of the PDF letter from the backup bucket, sanitises it and moves the
+    sanitised version to the final letters bucket.
+    """
+    current_app.logger.info(f'Re-sanitising and uploading PDF for notification with id {notification_id}')
+
+    try:
+        pdf_content = s3download(
+            current_app.config['PRECOMPILED_ORIGINALS_BACKUP_LETTER_BUCKET_NAME'],
+            f'{notification_id}.pdf'
+        ).read()
+
+        sanitisation_details = sanitise_file_contents(
+            pdf_content,
+            allow_international_letters=allow_international_letters,
+            filename=filename
+        )
+
+        # Only files that have failed sanitisation have 'message' in the sanitisation_details dict
+        if sanitisation_details.get('message'):
```

Comment:
We had been storing letters where the redaction of the address block failed so that we could take a closer look at why this was happening - it doesn't affect the sending process. That might be something we no longer need, but I'll leave that for another PR.

Reply:
That's fair, thanks for explaining 👍.